### PR TITLE
Issue #13345: Enabled InnerTypeLastCheckExamplesTest

### PR DESCRIPTION
**Pull request by MANISH-K-07**
Part of #13345 

Enabled InnerTypeLastCheckExamplesTest

### CODE REVIEW

**Comment by MANISH-K-07** [comment]
Github, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+*/
+package com.puppycrawl.tools.checkstyle.checks.design.innertypelast;
+
+// xdoc section -- start
+class Test1 {
+  private String s;
+  class InnerTest1 {}
+  public void test() {} // violation
+}
+
```
Comment (Vyom-Yadav) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java:16]: Why have you removed the explanation for the violation? 

```
// violation, method should be declared before inner types.
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class Test2 {
+  static {};
+  class InnerTest1 {}
+  public Test2() {} // violation
+}
```
Comment (Vyom-Yadav) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java:21]: Same here.

**Review by Vyom-Yadav — CHANGES_REQUESTED** [review]
Items:

**Comment by Vyom-Yadav** [comment]
I imagine what kind of CI these projects are using. I don't think the tests were even compiled.
`
Failed during checkstyle configuration: Exception was thrown while processing /home/circleci/project/.ci-temp/contribution/checkstyle-tester/src/main/java/cassandra/examples/ssl-factory/test/unit/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactoryTest.java: IllegalStateException occurred while parsing file /home/circleci/project/.ci-temp/contribution/checkstyle-tester/src/main/java/cassandra/examples/ssl-factory/test/unit/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactoryTest.java. 117:55: mismatched input ';' expecting ')': InputMismatchException
`
https://github.com/apache/cassandra/blob/997302d8a9a9b89f8e7fbc1821d693e04d314a66/examples/ssl-factory/test/unit/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactoryTest.java#L117-L118

@MANISH-K-07 There are multiple compilation errors in different projects, which are failing our CI. This is not new. We depend on multiple projects to test our changes. We clone those projects and run checkstyle against them to verify the changes. Can you send PRs in these projects to unblock our CI? The fixes are fairly simple.

**Comment by Vyom-Yadav** [comment]
You can use sourcegraph search for easier navigation.

**Comment by MANISH-K-07** [comment]
>  Can you send PRs in these projects to unblock our CI? The fixes are fairly simple.


@Vyom-Yadav ,
The failing CI have been resolved today morning. 
Now all the CI seem fine. Restarting the checks should do
This PR is a day old. Let me rebase and check....

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java (hunk 1 above)
Comment (MANISH-K-07) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java]: Sorry about that....

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java (hunk 2 above)
Comment (MANISH-K-07) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java]: Done
👍🏻

**Comment by MANISH-K-07** [comment]
These don't seem to be the issues caused by my input files ?....
Must have got tagged along with rebase ( affecting other PR #14428 also )

```
[INFO] There are 11 errors reported by Checkstyle 10.14.0-SNAPSHOT with checkstyle.xml ruleset.
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferenceEntryStore.java:[100,5] (metrics) CyclomaticComplexity: Cyclomatic Complexity is 14 (max allowed is 10).
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferenceEntryStore.java:[100,5] (metrics) NPathComplexity: NPath Complexity is 360 (max allowed is 200).
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferenceEntryStore.java:[100,5] (sizes) ExecutableStatementCount: Executable statement count is 41 (max allowed is 30).
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferenceEntryStore.java:[121,56] (coding) MultipleStringLiterals: The String "XWiki" appears 2 times in the file.
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferenceEntryStore.java:[121,65] (coding) MultipleStringLiterals: The String "Notifications" appears 2 times in the file.
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferenceEntryStore.java:[121,82] (coding) MultipleStringLiterals: The String "Code" appears 2 times in the file.
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferenceEntryStore.java:[170,69] (blocks) EmptyBlock: Empty if block.
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferencePropertyDescriptorStore.java:[45,5] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferencePropertyDescriptorStore.java:[47,5] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferencePropertyDescriptorStore.java:[49,5] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/main/java/org/xwiki/notifications/filters/internal/livedata/NotificationFilterPreferencePropertyDescriptorStore.java:[51,5] (javadoc) JavadocVariable: Missing a Javadoc comment.
```

**Comment by romani** [comment]
Looks like xwiki leaked regression in checkstyle validation
We need to put ` git checkout SHAbefore` on commit before they did latest update on such files

**Comment by MANISH-K-07** [comment]
GitHub, generate website

**Comment by MANISH-K-07** [comment]
> https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/dfcbef8_2024024524/index.html
> 
> https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/dfcbef8_2024024524/checks/design/innertypelast.html#Example1-code


@romani , Please review.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.design.innertypelast;
+
+// xdoc section -- start
+class Test1 {
+  private String s;
+  class InnerTest1 {}
+  public void test() {} // violation
+  // violation above, method should be declared before inner types.
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java:15]: We need only one comment about violation , remove `// violation`

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java (hunk 3 above)
Comment (MANISH-K-07) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/innertypelast/Example1.java]: I changed to single violation comment....
Please review if this is all good

**Comment by MANISH-K-07** [comment]
Github, generate website

File: src/xdocs/checks/design/innertypelast.xml [hunk 4]
```diff
@@ -29,22 +29,23 @@
         </source>
         <p id="Example1-code">Example:</p>
         <source>
-class Test {
-    private String s; // OK
-    class InnerTest1 {}
-    public void test() {} // violation, method should be declared before inner types.
+// violation message - method should be declared before inner types.
```
Comment (romani) [inline src/xdocs/checks/design/innertypelast.xml:32]: please remove this line.

you have vilation comment on lines where checkstyle place them, it is enough.

**Review by romani — CHANGES_REQUESTED** [review]
last item:

File: src/xdocs/checks/design/innertypelast.xml (hunk 4 above)
Comment (MANISH-K-07) [reply src/xdocs/checks/design/innertypelast.xml]: Sorry about that....
Done

**Comment by MANISH-K-07** [comment]
Github, generate website

**Comment by MANISH-K-07** [comment]
[Semaphore](https://checkstyle.semaphoreci.com/workflows/c70b2b34-277c-4887-a2f4-da9d132e1b55?pipeline_id=2d6e04ac-3b68-43e9-be42-795fdbd024b5) is all green, there are no errors but still shows as failure ?

**Review by romani — APPROVED** [review]
ok to merge.

@Vyom-Yadav , please finalize

**Comment by MANISH-K-07** [comment]
> ok to merge.
> 
> @Vyom-Yadav , please finalize

@romani ,
I think @Vyom-Yadav has been inactive since long

**Pull request merged**